pkg/controllers/dnspolicy/dns_helper.go

```diff
@@ -170,7 +230,7 @@ func (s *Service) GetDNSRecord(ctx context.Context, subDomain string, managedZon
 // ab2.lb-a1b2.shop.example.com A 192.22.2.3
 // ab3.lb-a1b2.shop.example.com A 192.22.2.4
 
-func (s *Service) SetEndpoints(ctx context.Context, mcgTarget *MultiClusterGatewayTarget, dnsRecord *v1alpha1.DNSRecord) error {
+func (dh *dnsHelper) setEndpoints(ctx context.Context, mcgTarget *dns.MultiClusterGatewayTarget, dnsRecord *v1alpha1.DNSRecord, listener *gatewayv1beta1.Listener) error {
```

Comment:
Doesn't look like listener is used,  i guess this is in prep for the next change.